indico/migrations/versions/20191002_1820_2496c4adc7e9_add_track_principal.py

```diff
@@ -0,0 +1,231 @@
+            u'NOT full_access', name=op.f('ck_track_principals_no_full_access')
+        ),
+        sa.CheckConstraint(
+            u'NOT read_access', name=op.f('ck_track_principals_no_read_access')
+        ),
+        sa.CheckConstraint(
+            u'email IS NULL OR email = lower(email)',
+            name=op.f('ck_track_principals_lowercase_email'),
+        ),
+        sa.CheckConstraint(
+            u'read_access OR full_access OR array_length(permissions, 1) IS NOT NULL',
+            name=op.f('ck_track_principals_has_privs'),
+        ),
+        sa.CheckConstraint(
+            u'type != 1 OR (email IS NULL AND event_role_id IS NULL AND local_group_id IS NULL AND '
+            'mp_group_name IS NULL AND mp_group_provider IS NULL AND user_id IS NOT NULL)',
+            name=op.f('ck_track_principals_valid_user'),
+        ),
+        sa.CheckConstraint(
+            u'type != 2 OR (email IS NULL AND event_role_id IS NULL AND mp_group_name IS NULL AND '
+            'mp_group_provider IS NULL AND user_id IS NULL AND local_group_id IS NOT NULL)',
+            name=op.f('ck_track_principals_valid_local_group'),
+        ),
+        sa.CheckConstraint(
+            u'type != 3 OR (email IS NULL AND event_role_id IS NULL AND local_group_id IS NULL AND '
+            'user_id IS NULL AND mp_group_name IS NOT NULL AND mp_group_provider IS NOT NULL)',
+            name=op.f('ck_track_principals_valid_multipass_group'),
+        ),
+        sa.CheckConstraint(
+            u'type != 4 OR (event_role_id IS NULL AND local_group_id IS NULL AND mp_group_name IS NULL AND '
+            'mp_group_provider IS NULL AND user_id IS NULL AND email IS NOT NULL)',
+            name=op.f('ck_track_principals_valid_email'),
+        ),
+        sa.CheckConstraint(
+            u'type != 6 OR (email IS NULL AND local_group_id IS NULL AND mp_group_name IS NULL AND '
+            'mp_group_provider IS NULL AND user_id IS NULL AND event_role_id IS NOT NULL)',
+            name=op.f('ck_track_principals_valid_event_role'),
+        ),
+        sa.ForeignKeyConstraint(
+            ['event_role_id'],
+            [u'events.roles.id'],
+            name=op.f('fk_track_principals_event_role_id_roles'),
+        ),
+        sa.ForeignKeyConstraint(
+            ['local_group_id'],
+            [u'users.groups.id'],
+            name=op.f('fk_track_principals_local_group_id_groups'),
+        ),
+        sa.ForeignKeyConstraint(
+            ['track_id'],
+            [u'events.tracks.id'],
+            name=op.f('fk_track_principals_track_id_tracks'),
+        ),
+        sa.ForeignKeyConstraint(
+            ['user_id'],
+            [u'users.users.id'],
+            name=op.f('fk_track_principals_user_id_users'),
+        ),
+        sa.PrimaryKeyConstraint('id', name=op.f('pk_track_principals')),
+        schema='events',
+    )
+    op.create_index(
+        op.f('ix_track_principals_email'),
+        'track_principals',
+        ['email'],
```

Comment:
all those simple single-column indexes can be written as `index=True` on the corresponding column definition above